shared/form-utils.js

```diff
@@ -0,0 +1,101 @@
+export function validateEmailInput(inputEl, errorEl) {
+  const value = inputEl.value;
+  const emailRegex = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
+
+  let message = "";
+
+  if (!value) {
+    message = "이메일을 입력해주세요.";
+  } else if (!emailRegex.test(value)) {
+    message = "잘못된 이메일 형식입니다 .";
+  }
+
+  showValidation(inputEl, errorEl, message);
+  return !message;
+}
+
+export function validateNicknameInput(inputEl, errorEl) {
+  const value = inputEl.value;
+
+  let message = "";
+  if (!value) {
+    message = "닉네임을 입력해주세요.";
+  }
+
+  showValidation(inputEl, errorEl, message);
+  return !message;
+}
+
+export function validatePasswordInput(inputEl, errorEl) {
+  const value = inputEl.value;
+  let message = "";
+
+  if (!value) {
+    message = "비밀번호를 입력해주세요.";
+  } else if (value.length < 8) {
+    message = "비밀번호를 8자 이상 입력해주세요.";
+  }
+
+  showValidation(inputEl, errorEl, message);
+  return !message;
+}
+
+export function validateConfirmPasswordInput(
+  passwordInput,
+  confirmPasswordInput,
+  errorEl
+) {
+  const passwordValue = passwordInput.value;
+  const confirmPasswordValue = confirmPasswordInput.value;
+
+  let message = "";
+
+  if (!confirmPasswordValue) {
+    message = "";
+  } else if (confirmPasswordValue !== passwordValue) {
+    message = "비밀번호가 일치하지 않습니다.";
+  }
+  showValidation(confirmPasswordInput, errorEl, message);
+  return !message;
+}
+
+export function togglePasswordVisiblity(passwordInput, eyeIcon) {
+  if (!passwordInput || !eyeIcon) return;
+
+  const isPasswordVisible = passwordInput.type === "text";
+  passwordInput.type = isPasswordVisible ? "password" : "text";
+  eyeIcon.src = isPasswordVisible
+    ? "/images/btn_visibility_off.png"
+    : "/images/btn_visibility_on.png";
+}
```

Comment:
`isPasswordVisible` 변수처럼 삼항 연산자 부분도 변수로 만들어서 사용하면 더 깔끔할 것 같아요.

```suggestion
  const isPasswordVisible = passwordInput.type === 'text';
  const newType = isPasswordVisible ? 'password' : 'text';
  const eyeIconSrc = isPasswordVisible
    ? '/images/btn_visibility_off.png'
    : '/images/btn_visibility_on.png';

  passwordInput.type = newType;
  eyeIcon.src = eyeIconSrc;
```